src/pip/_internal/req/req_file.py

```diff
@@ -68,7 +68,7 @@
 ]  # type: List[Callable[..., optparse.Option]]
 
 # the 'dest' string values
-SUPPORTED_OPTIONS_REQ_DEST = [o().dest for o in SUPPORTED_OPTIONS_REQ]
+SUPPORTED_OPTIONS_REQ_DEST = [str(o().dest) for o in SUPPORTED_OPTIONS_REQ]
```

Comment:
I'm very slightly concerned about this line but I can't really think of any reason this won't work.

Reply:
The problem seems to be that `optparse.Option.dest` can be of any type:

```python
>>> o = optparse.Option("-d", dest="foo")
>>> o.dest
'foo'
>>> o = optparse.Option("-d", dest=12)
>>> o.dest
12
```

Presumably mypy is failing to track the fact that we only supply strings to the "dest" of our options, so this is somehow trying to enforce that we do get a string here. (Or maybe there's a genuine bug and we do assign a non-string to a `dest` attribute somewhere?)

I'm slightly uncomfortable about adding a runtime call to `str()` just to placate mypy, but I'm not going to make a fuss about it. I would rather than there was a typecheck-time annotation we could use to say "don't make a fuss, this is always a string", though. This is the sort of busywork fix that I don't like about static checking...

Reply:
Cool. I'll revert this specific change and add an ignore instead in a follow up PR.